app/src/main/java/ch/epfl/sdp/contamination/ConcreteCachingDataSender.java

```diff
@@ -46,33 +56,42 @@ void setInteractor(GridFirestoreInteractor interactor) {
 
     @Override
     public void registerLocation(Carrier carrier, Location location, Date time) {
-        refreshLastPositions(time,location);
-        gridInteractor.write(location, String.valueOf(time.getTime()), carrier, successListener, failureListener);
+        registerLocation(carrier, location, time, successListener, failureListener);
     }
+
     @Override
     public void registerLocation(Carrier carrier,
                                  Location location,
                                  Date time,
                                  OnSuccessListener successListener,
-                                 OnFailureListener failureListener){
-        refreshLastPositions(time,location);
-        gridInteractor.write(location, String.valueOf(time.getTime()), carrier, successListener, failureListener);
+                                 OnFailureListener failureListener) {
+        refreshLastPositions(time, location);
 
+        Map<String, Object> element = new HashMap<>();
+        element.put("geoPoint", new GeoPoint(location.getLatitude(), location.getLongitude()));
+        element.put("timeStamp", time.getTime());
+        element.put("infectionStatus", carrier.getInfectionStatus());
+        gridInteractor.writeDocumentWithID("LastPositions", AccountFragment.getAccount(getActivity()).getId(),
```

Comment:
I think the carrier should have the same ID as the user ID... so no need to call AccountFragment.getAccount(getActivity()).getId() but just carrier.getUniqueID() or something

Reply:
Ok according to getUniqueId() method this returns "This identifier should NOT be the account ID since it's only used to distinguish between different carriers". You should think about which ID we want to store on firebase.

Reply:
I want the getID because getUniqueId is currently not implemented and return a single string "not unique" or somethind like that

Reply:
Mmmh yes that may be true. But with your method right now, you won't be able to post on firebase positions from different people during the tests
whereas, the getUniqueId() returns an ID that can be explicitely choosen when doing tests.
But it is not only about tests : on the real life, we don't wan't to store one's private ID publicly, so we use the carrier's ID.
You are true this ID is not yet implemented. We should do that soon, but it will work fine if you use this method